File: midonet-client/src/main/java/org/midonet/client/resource/PortGroup.java
/*
 * Copyright 2014 Midokura SARL
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.midonet.client.resource;

import java.net.URI;
import java.util.UUID;

import org.midonet.client.WebResource;
import org.midonet.client.dto.DtoPortGroup;

import static org.midonet.cluster.services.rest_api.MidonetMediaTypes.APPLICATION_PORTGROUP_JSON;

public class PortGroup extends ResourceBase<PortGroup, DtoPortGroup> {


    public PortGroup(WebResource resource, URI uriForCreation,
                     DtoPortGroup pg) {
        super(resource, uriForCreation, pg, APPLICATION_PORTGROUP_JSON());
    }

    @Override
    public URI getUri() {
        return principalDto.getUri();
    }

    public UUID getId() {
        return principalDto.getId();
    }

    public String getName() {
        return principalDto.getName();
    }

    public String getTenantId() {
        return principalDto.getTenantId();
    }

    public boolean isStateful() {
        return principalDto.isStateful();
    }

    public PortGroup name(String name) {
        principalDto.setName(name);
        return this;
    }

    public PortGroup tenantId(String tenantId) {
        principalDto.setTenantId(tenantId);
        return this;
    }

    public PortGroup stateful(boolean stateful) {
        principalDto.setStateful(stateful);
        return this;
    }

}

File: python-midonetclient/src/midonetclient/port_group.py
# vim: tabstop=4 shiftwidth=4 softtabstop=4

# Copyright 2013 Midokura PTE LTD.
# All Rights Reserved
#
# Licensed under the Apache License, Version 2.0 (the "License"); you may
# not use this file except in compliance with the License. You may obtain
# a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
# License for the specific language governing permissions and limitations
# under the License.


from midonetclient import port_group_port
from midonetclient import resource_base
from midonetclient import vendor_media_type
from vendor_media_type import APPLICATION_PORTGROUP_PORT_COLLECTION_JSON


class PortGroup(resource_base.ResourceBase):

    media_type = vendor_media_type.APPLICATION_PORTGROUP_JSON

    def __init__(self, uri, dto, auth):
        super(PortGroup, self).__init__(uri, dto, auth)

    def name(self, name):
        self.dto['name'] = name
        return self

    def tenant_id(self, tenant_id):
        self.dto['tenantId'] = tenant_id
        return self

    def get_name(self):
        return self.dto['name']

    def get_id(self):
        return self.dto['id']

    def get_tenant_id(self):
        return self.dto['tenantId']

    def is_stateful(self):
        return self.dto['stateful']

    def stateful(self, v):
        self.dto['stateful'] = v
        return self

    def get_ports(self, query=None):
        headers = {'Accept': APPLICATION_PORTGROUP_PORT_COLLECTION_JSON}
        return self.get_children(self.dto['ports'], query, headers,
                                 port_group_port.PortGroupPort)

    def add_port_group_port(self):
        return port_group_port.PortGroupPort(self.dto['ports'], {}, self.auth)
